pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -595,7 +595,11 @@ func (gc *GarbageCollector) GraphHasUID(UIDs []types.UID) bool {
 func GetDeletableResources(discoveryClient discovery.DiscoveryInterface) (map[schema.GroupVersionResource]struct{}, error) {
 	preferredResources, err := discoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get supported resources from server: %v", err)
+		if discovery.IsGroupDiscoveryFailedError(err) {
+			glog.V(0).Infof("failed to discover some groups: %v", err.(*discovery.ErrGroupDiscoveryFailed).Groups)
```

Comment:
TODO: decide on an appropriate log level.

Reply:
Should this log and tolerate *any* discovery error?

Reply:
Answer: yes

Reply:
Why not `glog.Warning`?

Reply:
And include a line to the effect that it will be retried.

Reply:
Fixed